cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -75,9 +80,33 @@
   private final String appClassName;
   private final Id.Artifact artifactId;
 
+  @Inject
+  public InMemoryConfigurator(CConfiguration cConf, PluginFinder pluginFinder, Impersonator impersonator,
+                              ArtifactRepository artifactRepository, @Assisted AppDeploymentInfo deploymentInfo) {
+    this.cConf = cConf;
+    this.pluginFinder = pluginFinder;
+    this.appNamespace = Id.Namespace.fromEntityId(deploymentInfo.getNamespaceId());
+    this.artifactId = Id.Artifact.fromEntityId(deploymentInfo.getArtifactId());
+    this.appClassName = deploymentInfo.getApplicationClass().getClassName();
+    this.applicationName = deploymentInfo.getApplicationName();
+    this.applicationVersion = deploymentInfo.getApplicationName();
+    this.configString = deploymentInfo.getConfigString() == null ? "" : deploymentInfo.getConfigString();
+    this.baseUnpackDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+                                  cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+
+    EntityImpersonator classLoaderImpersonator = new EntityImpersonator(deploymentInfo.getArtifactId(), impersonator);
+    ClassLoader tempLoader = null;
+    try {
+      tempLoader = artifactRepository.createArtifactClassLoader(deploymentInfo.getArtifactLocation(),
+                                                                classLoaderImpersonator);
+    } catch (IOException e) {
+      LOG.error("Failed to create ArtifactClassLoader for InMemoryConfigurator: ", e);
```

Comment:
Just log without failing? That's not a good practice. The creating of the classloader should be at the `config` method instead of in the constructor. Also, you need to close the classloader after using it (it is of type CloseableClassLoader).

Reply:
removed ability to pass in classLoader through constructor and moved classLoader creation to the config method. Also fixed error log without failing, it will not return a `Futures.immediateFailedFuture` if the class loader creation fails

Reply:
I reverted the change to the original constructor to maintain backwards compatibility, it caused failures in other places in the codebase.